### PR TITLE
Uar-1584 add extra noc

### PR DESCRIPTION
**Pull request by markpit**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-1584

### Change description
Adding new legal firm control NOC for BO individual Registration journey

### Work checklist

- [ ] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/utils/beneficial.owner.individual.ts [hunk 1]
```diff
@@ -184,21 +185,28 @@ export const setBeneficialOwnerData = (reqBody: any, id: string): ApplicationDat
   data[StartDateKey] = mapFieldsToDataObject(reqBody, StartDateKeys, InputDateKeys);
   data[CeasedDateKey] = reqBody["is_still_bo"] === '0' ? mapFieldsToDataObject(reqBody, CeasedDateKeys, InputDateKeys) : {};
 
+  setBoNocData(data);
```
Comment (mwestacott) [inline src/utils/beneficial.owner.individual.ts:188]: Logic placed in a separate local function to address the Sonar complexity violation.

File: src/utils/beneficial.owner.individual.ts [hunk 2]
```diff
@@ -184,21 +185,28 @@ export const setBeneficialOwnerData = (reqBody: any, id: string): ApplicationDat
...
+
+  data[ID] = id;
+
+  return data;
+};
+
+const setBoNocData = (data: ApplicationDataType) => {
   // It needs concatenations because if in the check boxes we select only one option
   // nunjucks returns just a string and with concat we will return an array.
   data[BeneficialOwnerNoc] = (data[BeneficialOwnerNoc]) ? [].concat(data[BeneficialOwnerNoc]) : [];
```
Comment (lduranteau) [inline src/utils/beneficial.owner.individual.ts:200]: Is it possible to change the template to return an array?

Comment (mwestacott) [reply src/utils/beneficial.owner.individual.ts:200]: Possibly, but I don't think we should do that as part of this change. We're simply following the same approach as we've done elsewhere.

**Review by dlloyd1-ch — APPROVED** [review]
LGTM

**Review by lduranteau — APPROVED** [review]

**Pull request merged**